### PR TITLE
fix(ProfileDescriptionPanel): Profile settings -> Identity : Addition of validation to bio field

### LINKED ISSUE #13623: Profile settings -> Identity : Addition of validation to bio field
Currently, the `Bio` field in `Profile` Settings does not have any validation meaning the user can add any character / emoji they desire.

It was recently confirmed that this field should allow only ASCII characters, not UTF-8 and should support emojis.

@alexjba had a couple of questions in regards to this so creating an issue for us to discuss / track decisions

Screenshot 2024-02-19 at 11 48 08 AM

Figma
https://www.figma.com/file/ibJOTPlNtIxESwS96vJb06/%F0%9F%91%A4-Profile-%7C-Desktop?type=design&node-id=2460-27368&mode=design&t=zk0Rc4XSIlXFhE07-4

### PR DESCRIPTION
**Pull request by Seitseman**
fixes #13623

### What does the PR do

- Added validator to the bio component to allow only ascii and emoji characters
- Added ProfileDescriptionPanelPage to a storybook

### Affected areas
Profile->views: ProfileDescriptionPanel, MyProfileView

image

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://github.com/status-im/status-desktop/assets/5157464/397939d2-4f29-4ff4-99ec-90c747ac2502

### CODE REVIEW

File: storybook/pages/ProfileDescriptionPanelPage.qml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                }
+                Label {
+                    id: customPrefName
+                    text: "Theo"
+                    enabled:emptyModelChecker.checked
+                }
+            }
+
+            CheckBox {
+                id: emptyModelChecker
```
Comment (caybro) [inline storybook/pages/ProfileDescriptionPanelPage.qml:114]: ?

File: storybook/pages/ProfileDescriptionPanelPage.qml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+                    implicitWidth: d.contentWidth
+
+                    readonly property bool isEnsName: profileStore.preferredName
+
+                    displayName.focus: !isEnsName
+                    displayName.input.edit.readOnly: isEnsName
+                    displayName.text: profileStore.name
+                    displayName.validationMode: StatusInput.ValidationMode.Always
+                    displayName.validators: isEnsName ? [] : Constants.validators.displayName
```
Comment (caybro) [inline storybook/pages/ProfileDescriptionPanelPage.qml:86]: I wonder why you have to change the behavior here in SB?

File: ui/imports/utils/Constants.qml [hunk 3]
```diff
@@ -681,6 +681,7 @@ QtObject {
...
...26a7\u26aa\u26ab\u26b0\u26b1\u26bd\u26be\u26c4\u26c5\u26c8\u26ce\u26cf\u26d1\u26d3\u26d4\u26e9\u26ea\u26f0-\u26f5\u26f7-\u26fa\u26fd\u2702\u2705\u2708-\u270d\u270f\u2712\u2714\u2716\u271d\u2721\u2728\u2733\u2734\u2744\u2747\u274c\u274e\u2753-\u2755\u2757\u2763\u2764\u2795-\u2797\u27a1\u27b0\u27bf\u2934\u2935\u2b05-\u2b07\u2b1b\u2b1c\u2b50\u2b55\u3030\u303d\u3297\u3299]|\ud83c[\udc04\udccf\udd70\udd71\udd7e\udd7f\udd8e\udd91-\udd9a\udde6-\uddff\ude01\ude02\ude1a\ude2f\ude32-\ude3a\ude50\ude51\udf00-\udf21\udf24-\udf93\udf96\udf97\udf99-\udf9b\udf9e-\udff0\udff3-\udff5\udff7-\udfff]|\ud83d[\udc00-\udcfd\udcff-\udd3d\udd49-\udd4e\udd50-\udd67\udd6f\udd70\udd73-\udd7a\udd87\udd8a-\udd8d\udd90\udd95\udd96\udda4\udda5\udda8\uddb1\uddb2\uddbc\uddc2-\uddc4\uddd1-\uddd3\udddc-\uddde\udde1\udde3\udde8\uddef\uddf3\uddfa-\ude4f\ude80-\udec5\udecb-\uded2\uded5-\uded7\udedc-\udee5\udee9\udeeb\udeec\udef0\udef3-\udefc\udfe0-\udfeb\udff0]|\ud83e[\udd0c-\udd3a\udd3c-\udd45\udd47-\ude7c\ude80-\ude88\ude90-\udebd\udebf-\udec5\udece-\udedb\udee0-\udee8\udef0-\udef8])(?:(?:\ud83c[\udffb-\udfff])?(?:\ud83e[\uddb0-\uddb3])?(?:\ufe0f?\u200d(?:[\u2000-\u3300]|[\ud83c-\ud83e][\ud000-\udfff])\ufe0f?)?)*))*$/
```
Comment (alexjba) [inline ui/imports/utils/Constants.qml:684]: We'll need to keep it inline with mobile regex to make sure we don't have any mismatches between platforms.

Please have a look at this regex used on mobile:
https://github.com/status-im/status-mobile/pull/19562/files#diff-9e385a06538a147df92716d390b2b37d2a07d495b3733f1f16439251b6d9dc6dR14

Please add a comment about keeping this regex inline.

cc @mohsen-ghafouri

File: ui/app/AppLayouts/Profile/panels/ProfileDescriptionPanel.qml [hunk 4]
```diff
@@ -53,7 +53,12 @@ Item {
             minimumHeight: 108
             maximumHeight: 108
             input.verticalAlignment: TextEdit.AlignTop
-
+            validators: [
+                StatusRegularExpressionValidator {
+                    regularExpression: Constants.regularExpressions.asciiWithEmoji
+                    errorMessage: qsTr("Only letters, numbers, ASCII characters or emojis allowed")
```
Comment (alexjba) [inline ui/app/AppLayouts/Profile/panels/ProfileDescriptionPanel.qml:59]: The error message has been updated in Figma to remove the ASCII reference.
https://www.figma.com/design/ibJOTPlNtIxESwS96vJb06/👤-Profile-|-Desktop?node-id=2460-8755&t=7EUM0YB7om3UmscW-0

**Review by alexjba — COMMENTED** [review]
Sorry about this, the task description wasn't properly updated.
Please have a look at these comments.

**Review by micieslak — CHANGES_REQUESTED** [review]
Hey, some comments/questions there :)

The sb page generates warnings:
```
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot call method 'validate' of null
file:///home/michalc/workspace/status-desktop/storybook/pages/ProfileDescriptionPanelPage.qml:20:1: QML ProfileDescriptionPanelPage: Binding loop detected for property "globalUtils"
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
file:///home/michalc/workspace/status-desktop/ui/imports/utils/Utils.qml:37: TypeError: Cannot read property 'isCompressedPubKey' of null
file:///home/michalc/workspace/status-desktop/ui/imports/utils/Utils.qml:37: TypeError: Cannot read property 'isCompressedPubKey' of null
file:///home/michalc/workspace/status-desktop/ui/imports/utils/Utils.qml:37: TypeError: Cannot read property 'isCompressedPubKey' of null
file:///home/michalc/workspace/status-desktop/ui/imports/utils/Utils.qml:37: TypeError: Cannot read property 'isCompressedPubKey' of null
```
Additionally, it's kind of rule that single page tests single component on top level, there are two: `ProfileHeader` and `ProfileDescriptionPanel`

**Comment by Seitseman** [comment]
> Hey, some comments/questions there :)
> 
> The sb page generates warnings:
> 
> ```
> file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
> file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
> file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
> file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
> file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
> file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
> file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
> file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot call method 'validate' of null
> file:///home/michalc/workspace/status-desktop/storybook/pages/ProfileDescriptionPanelPage.qml:20:1: QML ProfileDescriptionPanelPage: Binding loop detected for property "globalUtils"
> file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
> file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Controls/StatusInput.qml:285: TypeError: Cannot read property 'validate' of null
> file:///home/michalc/workspace/status-desktop/ui/imports/utils/Utils.qml:37: TypeError: Cannot read property 'isCompressedPubKey' of null
> file:///home/michalc/workspace/status-desktop/ui/imports/utils/Utils.qml:37: TypeError: Cannot read property 'isCompressedPubKey' of null
> file:///home/michalc/workspace/status-desktop/ui/imports/utils/Utils.qml:37: TypeError: Cannot read property 'isCompressedPubKey' of null
> file:///home/michalc/workspace/status-desktop/ui/imports/utils/Utils.qml:37: TypeError: Cannot read property 'isCompressedPubKey' of null
> ```
> 
> Additionally, it's kind of rule that single page tests single component on top level, there are two: `ProfileHeader` and `ProfileDescriptionPanel`

Those warnings are generated because some of the validators of the component use ```globalUtilsInst``` . It looks like other storybook pages have similar problems. Fix might be out of the scope of this task.

**Review by micieslak — COMMENTED** [review]
LGTM in general, some small notes only.

Moreover, the added page doesn't work properly right now:

Screenshot from 2024-05-20 10-50-29

It can be relatively easily fixed by mocking this `isAlias` method. Look e.g. for `MembersDropdownPage` where similar mocking is done. The long term solution is to remove access to backend from singletons, but for now workaround is fine. However the newly added page itself should work without errors/warnings imo.

**Review by micieslak — APPROVED** [review]
Nice!

**Review by caybro — APPROVED** [review]
LGTM :+1:

**Pull request merged**